activemq-unit-tests/src/test/java/org/apache/activemq/broker/jmx/BrokerViewTest.java

```diff
@@ -0,0 +1,68 @@
+ * limitations under the License.
+ */
+package org.apache.activemq.broker.jmx;
+
+import jakarta.jms.*;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.broker.BrokerRegistry;
+import org.apache.activemq.broker.BrokerService;
+import org.apache.activemq.util.Wait;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public class BrokerViewTest {
+    @Test(timeout=120000)
+    public void testBrokerViewRetrieveQueuesAndTopicsCount() throws Exception {
+        BrokerService brokerService = new BrokerService();
+        brokerService.setPersistent(false);
+        brokerService.start();
+
+        ActiveMQConnectionFactory factory =  new ActiveMQConnectionFactory(BrokerRegistry.getInstance().findFirst().getVmConnectorURI());
+        Connection producerConnection = factory.createConnection();
+        producerConnection.start();
+        // Create non-suppressed queue
+        Session producerSession = producerConnection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Queue queue = producerSession.createQueue("testQueue");
+        MessageProducer producer = producerSession.createProducer(queue);
+        producer.send(producerSession.createTextMessage("testMessage"));
+        // Create temporary queue
+        Session tempProducerSession = producerConnection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Queue tempQueue = tempProducerSession.createTemporaryQueue();
+        MessageProducer tempProducer = tempProducerSession.createProducer(tempQueue);
+        tempProducer.send(tempProducerSession.createTextMessage("testMessage"));
+        // Create non-suppressed topic
+        Session topicProducerSession = producerConnection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Topic topic = topicProducerSession.createTopic("testTopic");
+        MessageProducer topicProducer = topicProducerSession.createProducer(topic);
+        topicProducer.send(topicProducerSession.createTextMessage("testMessage"));
+        // Create temporary topic
+        Session tempTopicProducerSession = producerConnection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        Topic tempTopic = tempTopicProducerSession.createTemporaryTopic();
+        MessageProducer tempTopicProducer = tempTopicProducerSession.createProducer(tempTopic);
+        tempTopicProducer.send(tempTopicProducerSession.createTextMessage("testMessage"));
+
+        assertTrue(Wait.waitFor(() -> (brokerService.getAdminView()) != null));
+        final BrokerView view = brokerService.getAdminView();
+        assert(view.getTotalTopicsCount() == 11); // Including advisory topics
+        assert(view.getTotalNonSuppressedTopicsCount() == 11);
+        assert(view.getTotalTemporaryTopicsCount() == 1);
+        assert(view.getTotalQueuesCount() == 1);
+        assert(view.getTotalNonSuppressedQueuesCount() == 1);
+        assert(view.getTotalTemporaryQueuesCount() == 1);
```

Comment:
1. Please use the more appropriate assert* methods. 

2. These tests are dodgy. The value expect is 1 and the same across the different types. 

3. Multiple commits need to be squashed for this change

Reply:
I'll just point out that squashing commits is fine but it is also not necessary. I know I've mentioned before but I prefer it if people do not force push and just push new commits so you can see the difference easier between versions. It's easy to do a squash merge several commits at the end into 1 commit when merging back to main.